expr/expr_test.go

```diff
@@ -2102,13 +2108,159 @@ func TestEvalExpression(t *testing.T) {
 				args: []*expr{
 					{target: "metric1"},
 					{valStr: "si", etype: etString},
-					{valStr: "carrots", etype: etString},
 				},
 			},
 			map[MetricRequest][]*MetricData{
-				MetricRequest{"metric1", 0, 1}: {makeResponse("metric1", []float64{1, 2, 3, 4, 5}, 1, now32)},
+				MetricRequest{"metric1", 0, 1}: {
+					makeResponse("metric1",
+						[]float64{1.432729, 1.434207, 1.404762, 1.414609, 1.399159, 1.411343, 1.406217, 1.407123, 1.392078, math.NaN()}, 1, now32),
+				},
+			},
+			[]*MetricData{
+				makeResponse("metric1 Current: 1 Max: 1 Min: 1",
+					[]float64{1.432729, 1.434207, 1.404762, 1.414609, 1.399159, 1.411343, 1.406217, 1.407123, 1.392078, math.NaN()}, 1, now32),
+			},
+		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+					{valStr: "carrot", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {
+					makeResponse("metric1",
+						[]float64{1.432729, 1.434207, 1.404762, 1.414609, 1.399159, 1.411343, 1.406217, 1.407123, 1.392078, math.NaN()}, 1, now32),
+				},
+			},
+			[]*MetricData{
+				makeResponse("metric1 Current: 1 carrot Max: 1 carrot Min: 1 carrot",
+					[]float64{1.432729, 1.434207, 1.404762, 1.414609, 1.399159, 1.411343, 1.406217, 1.407123, 1.392078, math.NaN()}, 1, now32),
+			},
+		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {
+					makeResponse("metric1",
+						[]float64{math.NaN(), 88364212.53333333, 79008410.93333334, 80312920.0, 69860465.2, 83876830.0, 80399148.8, 90481297.46666667, 79628113.73333333, math.NaN()}, 1, now32),
+				},
+			},
+			[]*MetricData{
+				makeResponse("metric1 Current: 80M Max: 90M Min: 70M",
+					[]float64{math.NaN(), 88364212.53333333, 79008410.93333334, 80312920.0, 69860465.2, 83876830.0, 80399148.8, 90481297.46666667, 79628113.73333333, math.NaN()}, 1, now32),
+			},
+		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {
+					makeResponse("metric1",
+						[]float64{1000}, 1, now32),
+				},
+			},
+			[]*MetricData{
+				makeResponse("metric1 Current: 1k Max: 1k Min: 1k",
+					[]float64{1000}, 1, now32),
+			},
+		},
+		{
```

Comment:
Duplicate test for cactiStyle(metric1,"si") -> "metric1 Current: 1k Max: 1k Min: 1k"